src/freenet/io/comm/MessageFilter.java

```diff
@@ -46,8 +45,8 @@
     private boolean _matched;
     private PeerContext _droppedConnection;
     private MessageType _type;
-    private final HashMap<String, Object> _fields = new HashMap<String, Object>();
-    private final List<String> _fieldList = new LinkedList<String>();
+    private final List<Object> _fields = new ArrayList<Object>();
```

Comment:
I think a `LinkedHashMap` would be more appropriate here; it avoids some of the costs of a `HashMap` but helps our code to stay cleaner.